axelor-human-resource/src/main/java/com/axelor/apps/hr/service/project/ProjectTaskPPTGenerateServiceImpl.java

```diff
@@ -0,0 +1,148 @@
+package com.axelor.apps.hr.service.project;
+
+import com.axelor.apps.base.AxelorException;
+import com.axelor.apps.base.db.Unit;
+import com.axelor.apps.hr.db.Employee;
+import com.axelor.apps.hr.exception.HumanResourceExceptionMessage;
+import com.axelor.apps.project.db.ProjectPlanningTime;
+import com.axelor.apps.project.db.ProjectTask;
+import com.axelor.apps.project.service.app.AppProjectService;
+import com.axelor.auth.db.User;
+import com.axelor.common.ObjectUtils;
+import com.axelor.i18n.I18n;
+import com.google.inject.Inject;
+import com.google.inject.persist.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.commons.collections.CollectionUtils;
+
+public class ProjectTaskPPTGenerateServiceImpl implements ProjectTaskPPTGenerateService {
+  protected AppProjectService appProjectService;
+  protected ProjectPlanningTimeComputeService projectPlanningTimeComputeService;
+  protected ProjectPlanningTimeCreateService projectPlanningTimeCreateService;
+
+  @Inject
+  public ProjectTaskPPTGenerateServiceImpl(
+      AppProjectService appProjectService,
+      ProjectPlanningTimeComputeService projectPlanningTimeComputeService,
+      ProjectPlanningTimeCreateService projectPlanningTimeCreateService) {
+    this.appProjectService = appProjectService;
+    this.projectPlanningTimeComputeService = projectPlanningTimeComputeService;
+    this.projectPlanningTimeCreateService = projectPlanningTimeCreateService;
+  }
+
+  @Override
+  public String getSprintOnChangeWarningWithoutSprint(ProjectTask projectTask) {
+    BigDecimal oldBudgetedTime = projectPlanningTimeCreateService.getOldBudgetedTime(projectTask);
+
+    Set<ProjectPlanningTime> projectPlanningTimeSet =
+        getProjectPlanningTimeOnOldDuration(
+            projectTask, projectTask.getTaskDate(), oldBudgetedTime);
+
+    if (ObjectUtils.isEmpty(projectPlanningTimeSet)) {
+      if (ObjectUtils.isEmpty(projectTask.getProjectPlanningTimeList())) {
+        return I18n.get(HumanResourceExceptionMessage.PROJECT_PLANNING_TIME_FIRST_REQUEST);
+      } else {
+        return I18n.get(HumanResourceExceptionMessage.PROJECT_PLANNING_TIME_NEW_REQUEST);
+      }
+    } else {
+      if (projectPlanningTimeSet.size() == 1)
+        return I18n.get(HumanResourceExceptionMessage.PROJECT_PLANNING_TIME_UPDATE);
+      return I18n.get(HumanResourceExceptionMessage.PROJECT_PLANNING_TIME_EXISTING_ON_DURATION);
+    }
+  }
+
+  @Override
+  @Transactional
```

Comment:
```suggestion
  @Transactional(rollbackOn = {Exception.class})
```

else it will not rollback when `AxelorException` is thrown